contribs/gmf/src/directives/authenticationdirective.js

```diff
@@ -0,0 +1,137 @@
+goog.provide('gmf.AuthenticationController');
+goog.provide('gmf.authenticationDirective');
+
+goog.require('gmf');
+goog.require('gmf.Authentication');
+
+
+gmfModule.value('gmfAuthenticationTemplateUrl',
+    /**
+     * @param {angular.JQLite} element Element.
+     * @param {angular.Attributes} attrs Attributes.
+     */
+    function(element, attrs) {
+      var templateUrl = attrs['gmfAuthenticationTemplateurl'];
+      return templateUrl !== undefined ? templateUrl :
+          gmf.baseTemplateUrl + '/authentication.html';
+    });
+
+
+/**
+ * An "authentication" directive for a GeoMapFish application.
+ *
+ * @example
+ * <gmf-authentication></gmf-authentication>
+ *
+ * @param {string} gmfAuthenticationTemplateUrl Url to template.
+ * @return {angular.Directive} The Directive Definition Object.
+ * @ngInject
+ * @ngdoc directive
+ * @ngname gmfAuthentication
+ */
+gmf.authenticationDirective = function(gmfAuthenticationTemplateUrl) {
+  return {
+    scope: {},
+    controller: 'GmfAuthenticationController',
+    controllerAs: 'ctrl',
+    templateUrl: gmfAuthenticationTemplateUrl
+  };
+};
+
+gmfModule.directive('gmfAuthentication', gmf.authenticationDirective);
+
+
+
+/**
+ * @param {angular.Scope} $scope The directive's scope.
+ * @param {gmf.Authentication} gmfAuthentication GMF Authentication service
+ * @constructor
+ * @ngInject
+ * @ngdoc controller
+ * @ngname GmfAuthenticationController
+ */
+gmf.AuthenticationController = function($scope, gmfAuthentication) {
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.isLoggedIn = false;
+
+  /**
+   * @type {gmf.Authentication}
+   * @private
+   */
+  this.gmfAuthentication_ = gmfAuthentication;
+
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  this.scope_.formData = {};
```

Comment:
No.
Put your data into the controller itself, and access them from the view with the `ctrl` that comes from `controllerAs: 'ctrl'`